homeassistant/components/group/entity.py

```diff
@@ -425,27 +465,47 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
         elif tr_state.attributes.get(ATTR_ASSUMED_STATE):
             self._assumed_state = True
 
-        num_on_states = len(self._on_states)
+        # If we do not have an on state for any domains
+        # we use None (which will be STATE_UNKNOWN)
+        if (num_on_states := len(self._on_states)) == 0:
+            self._state = None
+            return
+
         # If all the entity domains we are tracking
         # have the same on state we use this state
         # and its hass.data[REG_KEY].on_off_mapping to off
+        group_is_on = self.mode(self._on_off.values())
         if num_on_states == 1:
             on_state = list(self._on_states)[0]
```

Comment:
Instead of the useless creation of a list, we can do:
```suggestion
            on_state = next(iter(self._on_states))
```

Not sure if it's faster though

Reply:
It's noticably faster:
```
>>> import timeit
>>> timeit.timeit("s = {1, 2, 3, 5, 6, 7, 8, 11, 13, 15}; next(iter(s))", number=1000000)
0.13482437503989786
>>> timeit.timeit("s = {1, 2, 3, 5, 6, 7, 8, 11, 13, 15}; list(s)[0]", number=1000000)
0.23214745800942183
```